demos/jobs_for_aws/jobs_demo.c

```diff
@@ -0,0 +1,789 @@
+                                   &pcJobId,
+                                   &usJobIdLength );
+
+        if( xStatus == JobsSuccess )
+        {
+            /* Upon successful return, the messageType has been filled in. */
+            if( ( topicType == JobsStartNextSuccess ) || ( topicType == JobsNextJobChanged ) )
+            {
+                /* Handler function to process payload. */
+                prvStartNextJobHandler( pxDeserializedInfo->pPublishInfo );
+            }
+            else if( topicType == JobsUpdateSuccess )
+            {
+                LogInfo( ( "Job update status request has been accepted by AWS Iot Jobs service." ) );
+            }
+            else if( topicType == JobsStartNextFailed )
+            {
+                LogWarn( ( "Request for next job description rejected: RejectedResponse=%.*s.",
+                           pxDeserializedInfo->pPublishInfo->payloadLength,
+                           ( const char * ) pxDeserializedInfo->pPublishInfo->pPayload ) );
+            }
+            else if( topicType == JobsUpdateFailed )
+            {
+                /* Set global flag to terminate demo as request for updating executing the job status
+                 * has been rejected by AWS IoT Jobs service. */
+                xDemoEncounteredError = pdTRUE;
+
+                LogWarn( ( "Request for job update rejected: RejectedResponse=%.*s.",
+                           pxDeserializedInfo->pPublishInfo->payloadLength,
+                           ( const char * ) pxDeserializedInfo->pPublishInfo->pPayload ) );
+
+                LogError( ( "Terminating demo as request to update job status has been rejected by "
+                            "AWS IoT Jobs service..." ) );
+            }
+            else
+            {
+                LogWarn( ( "Received an unexpected messages from AWS IoT Jobs service: "
+                           "JobsTopicType=%u", topicType ) );
+            }
+        }
+        else if( xStatus == JobsNoMatch )
+        {
+            LogWarn( ( "Incoming message topic does not belong to IoT Jobs!: topic=%.*s",
+                       pxDeserializedInfo->pPublishInfo->topicNameLength,
+                       ( const char * ) pxDeserializedInfo->pPublishInfo->pTopicName ) );
+        }
+        else
+        {
+            LogError( ( "Failed to parse incoming publish job. Topic=%.*s!",
+                        pxDeserializedInfo->pPublishInfo->topicNameLength,
+                        ( const char * ) pxDeserializedInfo->pPublishInfo->pTopicName ) );
+        }
+    }
+    else
+    {
+        vHandleOtherIncomingPacket( pxPacketInfo, usPacketIdentifier );
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Entry point of the Jobs demo.
+ *
+ * This main function demonstrates how to use the Jobs library API Device Shadow library.
+ *
+ * The helper functions this demo uses for MQTT operations have internal
```

Comment:
```suggestion
 * The helper functions this demo used for MQTT operations have internal
```

Reply:
I think the use of simple present tense makes sense in this statement as it is only describing the operations that the function _conducts_ (simple present)

Reply:
Oh ok I see that now, there is a word missing that made it read weird
```suggestion
 * The helper functions this demo uses for MQTT operations that have internal
```